srml/support/src/storage/mod.rs

```diff
@@ -442,13 +447,56 @@ where
 /// Note that `storage_key` must be unique and strong (strong in the sense of being long enough to
 /// avoid collision from a resistant hash function (which unique implies)).
 pub mod child {
-	use super::{Codec, Decode, Vec, IncrementalChildInput};
+	use super::{runtime_io, Codec, Decode, Vec, IncrementalChildInput, ChildTrie,
+		ChildTrieReadRef, well_known_keys};
+
+	/// Method for fetching or initiating a new child trie.
+	pub fn next_keyspace() -> u128 {
+		let key = well_known_keys::CHILD_STORAGE_KEYSPACE_COUNTER;
+		// do start at 1 (0 is reserved for top trie)
+		let previous = super::unhashed::get(key).unwrap_or(0u128);
+		let new = previous + 1;
+		super::unhashed::put(key, &new);
+		new
+	}
+
+	/// Method for fetching or initiating a new child trie.
+	pub fn fetch_or_new(
+		parent: &[u8],
+	) -> ChildTrie {
+		ChildTrie::fetch_or_new(
+			|pk| { child_trie(pk) },
+			|ct| {
+				let updated = set_child_trie(ct);
+				// fetch or new create a new child trie
+				// so the child trie creation condition
+				// cannot fail at this point.
+				debug_assert!(updated);
+			},
+			parent,
+			next_keyspace(),
```

Comment:
Provide this as a function so that the counter is not incremented when the child trie already exists.